convert2rhel/unit_tests/logger_test.py

```diff
@@ -73,9 +102,37 @@ def test_tools_opts_debug(tmpdir, read_std, is_py2):
 def test_logger_custom_logger(tmpdir, caplog):
     """Test CustomLogger."""
     log_fname = "convert2rhel.log"
-    logger_module.initialize_logger(log_name=log_fname, log_dir=str(tmpdir))
+    logger_module.setup_logger_handler(log_name=log_fname, log_dir=str(tmpdir))
     logger = logging.getLogger(__name__)
     logger.task("Some task")
     logger.file("Some task write to file")
     with pytest.raises(SystemExit):
         logger.critical("Critical error")
+
+
+@pytest.mark.parametrize(
+    ("log_name", "path_exists", "is_dir", "exception"),
+    (
+        ("convert2rhel.log", True, True, False),
+        ("convert2rhel.log", True, True, False),
+        ("convert2rhel.log", False, False, False),
+        ("convert2rhel.log", True, False, True),
+    ),
+)
+@mock.patch("os.stat")
+def test_archive_old_logger_files(os_stat, log_name, path_exists, is_dir, exception, monkeypatch, caplog):
+    path_exists_mock = mock.MagicMock(return_value=path_exists)
+    is_dir_mock = mock.MagicMock(return_value=is_dir)
+    shutil_mock = mock.MagicMock(return_value=None)
+
+    monkeypatch.setattr(os.path, "exists", value=path_exists_mock)
+    monkeypatch.setattr(os.path, "isdir", value=is_dir_mock)
+    monkeypatch.setattr(shutil, "copyfile", value=shutil_mock)
+
+    os_stat.return_value.st_mtime_ns = 1234
+
+    if exception:
+        with pytest.raises(IOError):
+            logger_module.archive_old_logger_files(log_name)
+    else:
+        assert logger_module.archive_old_logger_files(log_name) is None
```

Comment:
I suggest using pytest's tmpdir facility for this test instead of mocks.  This is what it would look like (Note: test won't pass/work until log_dir is added to `archive_old_logger_files()`: https://gist.github.com/abadger/19b31a1cd186c1dddd7351cba77b90a3

Using tmpdir pointed out that the mock based test would not have been able to test the case where current_log exists but the archive directory does not since checking for the current_log and archive directory are both done using os.path.exists(). (You could modify the mock based tests to return different values based on what path is passed in but that would make the mock based test even more complex)